Core/src/main/java/com/plotsquared/core/util/task/TaskManager.java

```diff
@@ -25,144 +25,227 @@
  */
 package com.plotsquared.core.util.task;
 
-import com.plotsquared.core.PlotSquared;
-import com.plotsquared.core.util.RuntimeExceptionRunnableVal;
 import lombok.Getter;
 import lombok.Setter;
 
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * Task manager that handles scheduling of tasks.
+ * Synchronous methods make no guarantee of being scheduled on the
+ * server thread, instead they guarantee that no two synchronous
+ * operations happen at the same time. Implementations of
+ * the task manager might make other guarantees. All asynchronous
+ * operations will happen on another thread, no matter where
+ * they're scheduled from.
+ */
 public abstract class TaskManager {
 
     public static final HashSet<String> TELEPORT_QUEUE = new HashSet<>();
-    public static final HashMap<Integer, Integer> tasks = new HashMap<>();
+    public static final HashMap<Integer, PlotSquaredTask> tasks = new HashMap<>();
     public static AtomicInteger index = new AtomicInteger(0);
 
-    @Getter @Setter private static TaskManager implementation;
-
-    public static int runTaskRepeat(Runnable runnable, int interval) {
```

Comment:
Do we still have the general plan to implement our own platform-agnostic Async stuff, for whatever might never use it?

Reply:
Maybe:tm: We currently don't need it, if we do, we implement it then. There's nothing stopping us. I also think it's fine if the static methods use the platform one, so we use the platform task manager by default.